indexer_worker/indexer_worker/api.py

```diff
@@ -116,6 +117,7 @@ def create_api():
     _api.add_route("/healthcheck", HealthcheckResource())
     _api.add_route("/task", IndexingJobResource(task_tracker))
     _api.add_route("/task/{task_id}", TaskStatusResource(task_tracker))
+    _api.add_static_route("/static", (Path(".") / "static").absolute())
```

Comment:
The indexer worker serves the mock sensitive terms list, similar to how the ingestion server currently does so. This is okay because in local development, the indexer worker is always running, and in production we use the *real* sensitive terms list rather than the mock one. It is helpful to have the mock list served this way to make the local `filtered_index` steps behave as similarly as possible to the way it does in production, just using a different endpoint.

Reply:
If we wanted to do it a different way (to avoid adding this functionality to the indexer worker), we could put the file into minio, and still retrieve it over HTTP. [According to this guide about using minio to host a website](https://gist.github.com/harshavardhana/f05b60fe6f96803743f38bea4b565bbf) it shouldn't be too difficult to do later on.

For the sake of simplicity, though, would it be better if the DAG just swapped the location it reads the sensitive terms from based on an additional environment variable? Like:

```py
if USE_MOCK_SENSITIVE_TERMS:
  return ["running", "water", "bird"]
else:
  # retrieve the regular list
```

I know it's not as seamless as just swapping out the HTTP connection string in the environment file, but it avoids a bit of this hoop-jumping required to make the DAG code "seamless". I guess really we've just hidden the seam somewhere else by doing it this way :stuck_out_tongue: (Also: because I wrote the original code for this in the ingestion server, I can say I think it was a bit cheeky/too clever to do it this way and wish I'd done this if/else there to begin with!)

Reply:
We couldn't use the HttpOperator if we wanted to do that, or else we'd have to add branching tasks to get around selecting the right task per environment. Not a big deal but I'd prefer to keep the operators clean. We _could_ use minio, something similar to what @AetherUnbound did in #4836 maybe 🤔

Reply:
> We couldn't use the HttpOperator if we wanted to do that, or else we'd have to add branching tasks to get around selecting the right task per environment

Oh :confused: We can't call the HttpOperator conditionally in a Python `@task` operator? I figured something like this would be possible:

```py
@task
def get_sensitive_terms(**context):
    if USE_MOCK_SENSITIVE_TERMS:
        return ["running", "water", "bird"]
    else:
        http_op = HttpOperator(...)
        return http_op.execute(**context)
```

Reply:
We might be able to initialize the operator inside a `@task`, but it's probably easier to use the `HttpHook` inside a `@task` than have an operator-in-operator situation by using `HttpOperator`.

I think I'm fine going with the new static route approach suggested here, but @stacimc do you mind adding a comment about what the route is for, and that it's only intended to be used for local testing?

Reply:
Ah, maybe I'm thinking of the hook.

All good, I was just suggesting alternatives _if wanted_, this approach obviously has worked fine for us :slightly_smiling_face: 

It will go away in #3336 anyway, which I didn't know when I wrote my original comment or I would have mentioned it!